pkg/integrations/manager.go

```diff
@@ -246,6 +256,13 @@ func (m *Manager) run(ctx context.Context) {
 }
 
 func (m *Manager) runIntegration(ctx context.Context, i common.Integration) {
+	defer func() {
+		if r := recover(); r != nil {
+			err := fmt.Errorf("%v", r)
+			level.Error(m.logger).Log("msg", "integration has panicked. THIS IS A BUG!", "err", err, "integration", i.Name())
+		}
+	}()
+
```

Comment:
Seems like there's a rare race condition when shutting down the Agent and shutting down the statsd integration. This makes it a non-fatal error, but since it's only happening on process shutdown, it's not a big deal for now anyway. I'll leave this as a log line that denotes it as a bug and we can follow up again later.